R/messages.R

```diff
@@ -110,3 +110,85 @@ setvar <- function(core, dates, var, values, unit)
 
     invisible(NULL)
 }
+
+
+#' Fetch all the available Hector variables
+#'
+#' This function is similar to fetchvars, except it fetches all available
+#' Hector variables, including exogenous inputs (e.g., emissions & pre-industrial
+#' concentrations), parameters (e.g., alpha, beta, ECS), and output
+#' variables (e.g., concentrations & forcings).
+#'
+#' @param core Hector core object
+#' @param dates Vector of dates, optional
+#' @param scenario Scenario name, optional str
+#' @param outpath Absolute path of the output csv. Output will only be written to
+#' csv if this is given
+#' @return vars_all    Dataframe containing all Hector variables
+#' @family main user interface functions
+#' @export
+fetchvars_all <- function(core, dates=NULL, scenario=NULL, outpath=NULL) {
```

Comment:
Not everybody agrees with that prescription.  There is something to be said for putting closely related terms close together and separating them from less closely related terms with spaces.  For example, to my eye `a*c - b*d` is more readable than `a * c - b * d`.  The goal should always be readability, not adherence to some arbitrary convention.

Personally, I think the argument list is more readable _without_ spaces around the `=` signs because it's easier to see at a glance the number and names of the arguments.

Reply:
Fair enough. I usually try to follow the [Tidyverse style guide's recommendations](https://style.tidyverse.org/syntax.html#infix-operators), which, AFAICT, is the most widely used R style guide. But I think the No.1 rule of every style guide is (or at least should be) "If breaking this style guide makes your code more readable, break it."

I've gotten pretty accustomed to seeing `parameter = value` with the spaces, so `parameter=value` to me looks a bit cramped and less readable 🤷‍♂

Reply:
The tidyverse developers are just these guys, you know?  We've disagreed with them about things in the past and probably will do again in the future, and that's ok.  As long as the code is readable, I don't think we should be dogmatic about spacing.  And it's even ok if the spacing varies a bit from place to place in the code, depending on who wrote it.

All those statements are just my opinions, of course.